vsintegration/src/vs/FsPkgs/FSharp.Project/Common.Source.CSharp/Project/ProjectConfig.cs

```diff
@@ -1625,11 +1655,60 @@ internal bool IsUpToDate(OutputWindowLogger logger, bool testing)
                     freshestInputTime = timeStamp.Value;
             }
 
+            // check 1-1 mappings
+            foreach (var kv in directMappings)
+            {
+                string input = kv.Item1;
+                string copyToOutputDirectory = kv.Item2;
+                string output = kv.Item3;
+
+                var inputTime = TryGetLastWriteTimeUtc(input, logger);
+                if (!inputTime.HasValue)
+                {
+                    logger.WriteLine("Declaring project NOT up to date, can't find expected input {0}", input);
+                    return false;
+                }
+
+                if (String.Compare(copyToOutputDirectory, "Never", StringComparison.OrdinalIgnoreCase) == 0)
+                {
+                    //No need to check output
```

Comment:
Inputs = items with itemType Compile != Content items

If i add a text file readme.md (Content item with Copy=Never):
- i trigger i build. the fsproj is changed -> build -> new .exe
- i change the readme.md
- i trigger another build. how can i know if the file is changed (using timestamps)? 

let's say i check the timestamp of Content item and .exe timestamp. 
The .exe timestamp is < than changed readme.md. so i say yes to build. 
But after the build the exe is unchanged. msbuild doesn't build the exe (the input of .exe are the compile items only and other special items like resources etc). 
So now what? From now on, every build has the same problem, readme.md timestamp > .exe timestamp